clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -333,4 +334,15 @@ static void checkException(IOException e, TestCloseable... closeablesWithExcepti
         }
     }
 
+    @Test(timeout = 120000)
+    public void testRecursiveDelete() throws IOException {
+        Utils.delete(null); // delete of null does nothing.
+        Utils.delete(TestUtils.tempFile()); // deleting a temporary file works
+        File tempDir = TestUtils.tempDirectory();
+        File tempDir2 = TestUtils.tempDirectory(tempDir.toPath(), "a");
+        TestUtils.tempDirectory(tempDir.toPath(), "b");
+        TestUtils.tempDirectory(tempDir2.toPath(), "c");
+        Utils.delete(tempDir); // deleting a directory hierarchy works
+        Utils.delete(tempDir); // deleting a non-existent directory hierarchy works
```

Comment:
In this test, should we verify that the file/directory is actually gone after deletion?